src/operators/semesterOperations.ts

```diff
@@ -0,0 +1,48 @@
+import Semester, { SemesterInterface } from '../models/semester';
+
+export const create = async (semester: SemesterInterface) => {
+	try {
+		await Semester.create(semester);
+	} catch (err) {
+		throw 'ERROR: semester not created';
+	}
+};
+
+// NOTE: any reference to key is referring to the hash key, or the `id` parameter in all the models.
+export const read = async (key: string) => {
+	// add some kind of security check here?
+	const semester = await Semester.get(key);
+	if (semester) {
+		return semester;
+	} else {
+		throw 'ERROR: could not read semester';
+	}
```

Comment:
Quick reminder about how await works: If the promise resolves, everything is fine and `semester` is initialized to the semester we tried fetching. If the promise rejects (failure for any reason) it will throw, and it will go to the nearest catch block to handle (or crash the program if there is no catch in the call stack).

So if the `await` call fails and a semester is not resolved it will throw. Given this, if we make it to line 15, we know before even checking the semester that it resolved. Since we are just returning the semester or throwing, like the await call does we can do the following:

```ts
export const read = (key:string) => Semester.get(key);
```

To elaborate more on the function above, the Semester.get returns a promise, and we are just retuning that promise immediately. Since the caller is calling a function that returns a promise, they are responsible for handling the failure if they want to do something with it. They can also choose to await or not.

Alternatively, if we wanted to handle the error in this function, we would try/catch here. But I don't think this is what we want to do.

Reply:
I tried this change but when I test a get request with a semester id that doesn't exist in the database Semester.get(key) does not fail and throw.

Reply:
Oh very interesting! What happens instead? Is the semester just undefined?

Reply:
Ya - just an empty body returned. Technically it returns: 
`{
  "courses": []
}`
but that is because we're converting the courses set to an array to be returned!

Reply:
Okay well I like your strategy of throwing if the semester is undefined in this case! Good testing 😄